src/utils/ic.rs

```diff
@@ -0,0 +1,23 @@
+use ic_agent::{agent::AgentBuilder, Agent, AgentError};
+
+use crate::consts::AGENT_URL;
+
+#[derive(Clone)]
+pub struct AgentWrapper(Agent);
+
+impl AgentWrapper {
+    pub fn build(builder_func: impl FnOnce(AgentBuilder) -> AgentBuilder) -> Self {
+        let mut builder = Agent::builder().with_url(AGENT_URL);
+        builder = builder_func(builder);
+        Self(builder.build().unwrap())
+    }
+
+    pub async fn get_agent(&self) -> Result<&Agent, AgentError> {
+        let agent = &self.0;
+        #[cfg(any(feature = "local-bin", feature = "local-lib"))]
+        {
+            agent.fetch_root_key().await?;
+        }
```

Comment:
any reason for doing this everytime and not just once in the build step ?

Reply:
internally it has a check which skips the fetch. Ideally i would have wanted to do this in a constructor as well but that'd end being a larger change and might've introduced some latency